csharp/src/Google.Protobuf.Conformance/Program.cs

```diff
@@ -45,6 +47,8 @@ class Program
     {
         private static void Main(string[] args)
         {
+            var mode = (args[0] == "--buffer") ? SerializationMode.Buffer : SerializationMode.Stream;
```

Comment:
we should make the command line parsing a bit more robust, like this it's really easy to unknowingly fallback to SerializationMode.Stream (e.g. when the cmdline arg is misspelled it will be just ignored) and thus skip testing buffer serialization completely.

the "--buffer" argument name is not very descriptive (how about `--buffer_serialization` or `--use_buffer_serialization`?)